packages/wrangler/src/pipelines/cli/create.ts

```diff
@@ -29,274 +16,132 @@ export const pipelinesCreateCommand = createCommand({
-		) {
-			// auto-generate a service token
-			const auth = await authorizeR2Bucket(
-				config,
-				name,
-				accountId,
-				pipelineConfig.destination.path.bucket
-			);
-			destination.credentials.access_key_id = auth.accessKeyId;
-			destination.credentials.secret_access_key = auth.secretAccessKey;
+		if (!args.sql && !args.sqlFile) {
+			throw new UserError("Either --sql or --sql-file must be provided");
 		}
-
-		if (!destination.credentials.access_key_id) {
-			throw new FatalError("Requires a r2 access key id");
+		if (args.sql && args.sqlFile) {
+			throw new UserError("Cannot specify both --sql and --sql-file");
 		}
 
-		if (!destination.credentials.secret_access_key) {
-			throw new FatalError("Requires a r2 secret access key");
+		let sql: string;
+		if (args.sql) {
+			sql = args.sql;
+		} else if (args.sqlFile) {
+			try {
+				sql = readFileSync(args.sqlFile, "utf-8").trim();
+			} catch (error) {
+				throw new UserError(
+					`Failed to read SQL file '${args.sqlFile}': ${error instanceof Error ? error.message : String(error)}`
+				);
+			}
+		} else {
+			// This should never happen due to earlier validation
+			throw new UserError("Either --sql or --sql-file must be provided");
 		}
 
-		if (args.source.length > 0) {
-			const sourceHandlers: Record<string, () => Source> = {
-				http: (): HttpSource => {
-					const http: HttpSource = {
-						type: "http",
-						format: "json",
-						authentication: args.requireHttpAuth,
-					};
-
-					if (args.corsOrigins && args.corsOrigins.length > 0) {
-						http.cors = { origins: args.corsOrigins };
-					}
-
-					return http;
-				},
-				worker: (): BindingSource => ({
-					type: "binding",
-					format: "json",
-				}),
-			};
+		if (!sql) {
+			throw new UserError("SQL query cannot be empty");
+		}
 
-			for (const source of args.source) {
-				const handler = sourceHandlers[source];
-				if (handler) {
-					pipelineConfig.source.push(handler());
+		// Validate SQL before creating pipeline
+		logger.log("🌀 Validating SQL...");
+		try {
+			const validationResult = await validateSql(config, { sql });
+
+			if (
+				validationResult.tables &&
+				Object.keys(validationResult.tables).length > 0
+			) {
+				const tableNames = Object.keys(validationResult.tables);
+				logger.log(
+					`✅ SQL validated successfully. References tables: ${tableNames.join(", ")}`
+				);
+			} else {
+				logger.log("✅ SQL validated successfully.");
+			}
+		} catch (error) {
+			let errorMessage = "Unknown validation error";
+			if (error && typeof error === "object") {
+				const errorObj = error as {
+					notes?: Array<{ text?: string }>;
+					message?: string;
+				};
+				if (
+					errorObj.notes &&
+					Array.isArray(errorObj.notes) &&
+					errorObj.notes[0]?.text
+				) {
+					errorMessage = errorObj.notes[0].text;
+				} else if (error instanceof Error) {
+					errorMessage = error.message;
 				}
 			}
+			throw new UserError(`SQL validation failed: ${errorMessage}`);
 		}
```

Comment:
This should not be needed. The API Error will be displayed by Wrangler and should have enough detail